### PR TITLE
Add section on reading Gaia data tables with astropy

### LINKED ISSUE #13331: ECSV fails to read null values
### Description

Loading an ECSV file with a float column that contains any "null" values results in an error: `ValueError: could not convert string to float: 'null'`.

This error is thrown at this line: https://github.com/astropy/astropy/blob/dd8ccdde27858d076b8dbb676364e520ca06f987/astropy/io/ascii/ecsv.py#L340

If you trace back the root of the error, it goes back to this line, where `numpy.array` does not convert "null" to a value (such as `numpy.nan`): https://github.com/astropy/astropy/blob/dd8ccdde27858d076b8dbb676364e520ca06f987/astropy/io/ascii/core.py#L1008

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to add documentation describing how to read Gaia DR3 data tables using astropy. Those data tables include ECSV headers but the data are not ECSV-compliant because they use `null` instead of `""` to indicate missing data.

The intent here is that this new section will be linked in the Gaia documentation and/or a google search will turn up this section.

Fixes #13331

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by eerovaher** [comment]
_Gaia_ data is used widely enough that documenting this workaround is a good idea, but I think it is also worth making it clear that the problem only concerns the bulk data file, and in many cases a better option would be to query only the relevant data from the _Gaia_ archive (with e.g. `astroquery`), in which case the described workaround would not needed.

**Comment by mhvk** [comment]
Looks good but like @eerovaher's suggestion. I think it can be as simple as linking to https://astroquery.readthedocs.io/en/latest/gaia/gaia.html

**Review by pllim — APPROVED** [review]
LGTM. Not sure why this took so long. I threw in some extra CI but really I want to make sure your new URL does not trip linkcheck. Thanks!

**Comment by pllim** [comment]
I also took the liberty to rebase and address review comment, since it needed a rebase to make RTD pass. FYI.

File: docs/io/ascii/read.rst [hunk 1]
```diff
@@ -1014,3 +1014,32 @@ just two lines the problem is easy to spot, but for longer tables, the line numb
...
 if it works or if there is a another badly formatted data line.
+
+.. _io-ascii-read-gaia-tables:
+
+Reading Gaia Data Tables
+========================
+
+.. note::
+
+    The recommended way to access Gaia is via its
```
Comment (pllim) [inline docs/io/ascii/read.rst:1025]: @eerovaher , hopefully this addresses your concern?

**Comment by pllim** [comment]
Bot where you at?

@meeseeksdev backport to v5.0.x

@meeseeksdev backport to v5.2.x

**Pull request merged**